### ISSUE TITLE
Broken regression testing on laravel/framework

### ISSUE THREAD

**Issue by GrahamCampbell** [FriendsOfPHP/PHP-CS-Fixer#6394]
~After merging #6380, #6384, #6388, #6391 into master,~ we're now making incorrect changes to various repos, such as `laravel/framework`. ~I will narrow this down more, shortly.~ (EDIT: Master is broken. It was caused by the merge of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5960.) Here's a couple of broken files:

```diff
diff --git a/src/Application.php b/src/Application.php
index 757d8345a9ab4702eec31aa9b7c785863d0c26b9..a918c748b7e3756da57ec02d68020292520b6e0e 100644
--- a/src/Application.php
+++ b/src/Application.php
@@ -46,7 +46,7 @@
 class Application extends Container
 {
     use Concerns\RoutesRequests,
-        Concerns\RegistersExceptionHandlers;
+    Concerns\RegistersExceptionHandlers;
 
     /**
      * Indicates if the class aliases have been registered.
```

```diff
diff --git a/src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php b/src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php
index fe5f54b79765089102fabc824144879e54ea0899..dbb8c932871230222bfed9f348a4c472908da640 100755
--- a/src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php
+++ b/src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php
@@ -64,8 +64,8 @@ class DatabaseTokenRepository implements TokenRepositoryInterface
      * @return void
      */
     public function __construct(ConnectionInterface $connection, HasherContract $hasher,
-                                $table, $hashKey, $expires = 60,
-                                $throttle = 60)
+        $table, $hashKey, $expires = 60,
+        $throttle = 60)
     {
         $this->table = $table;
         $this->hasher = $hasher;
```

**Comment by GrahamCampbell** [comment]
Ok, narrowed it down - it's none of those PRs. Master is already broken. It was caused by the merge of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5960.

**Comment by julienfalque** [comment]
I agree the first example should not happen, I'll try to have a look. However the second example looks legit IMO.

**Comment by GrahamCampbell** [comment]
The 2nd example is a breaking change to how the braces fixer works, and should be controlled with a config option.

**Comment by SpacePossum** [comment]
beside the BC argument, may I ask if you find the current CS better than the fixed one (of the 2nd example)?

**Comment by GrahamCampbell** [comment]
Laravel wants to keep using the braces fixer, but doesn't want that change to be applied, if that answers your question.

**Comment by SpacePossum** [comment]
>, if that answers your question.

Kind of, I mean I asked your opinion and I got the Laravel preference, so I'm still not sure if these are inline, but I appreciate the answer :)

The rework of the braces fixer is major, Julien already did a ton of work to split it so it will be more easy to control by end users (different rules and config etc.) and the maintenance level will be dropped a ton.
We cannot keep everything as it was, especially because the current fixer has it flaws pointed out here by many issues raised by users. Splitting it up means it becomes possible to fix older issues nobody could, wanted or dare to touch.
Now on this drawback you reported, the braces rule has (AFAIK) always stated it will fix according to PSR2. Now, sure granted, it not always did. However, this description of the working was always what was communicated as it purpose to end users.
So now, if Julien (or anyone else who has the time and energy for it) refines the rules(s) to work more to its PSR2 propose and promise this is not a BC break, it is closer to (if not completely) a bug fix IMHO. As such, I don't agree the 2nd example is a valid new bug, it is the result of an old bug squashed and I'm happy Julien did :+1:

**Comment by julienfalque** [comment]
Thanks @SpacePossum 😄

I do think the new behavior is closer to what PSR-12 expects. My plan regarding indentation is to have everything in a single rule. Currently we have `statement_indentation` and `array_indentation`. Since they work exactly the same, I'd like to merge both into a single rule in the future.

Even so, their implementation is complex so I would avoid adding options as much as possible and I wouldn't support ignoring indentation for method parameters in the new rules. However, when used internally by `braces`, `StatementIndentationFixer` has a BC compatibility mode to maintain `braces` behavior. Method parameters indentation is simply a case I missed, I don't think this is actually covered by `BracesFixerTest`.

I can try—if that's not too much work—to restore the previous behavior for `braces`. You won't have a "BC break" anymore for `braces`, but there won't be any replacement for this old behavior in the new rules. Does that sound ok to you @GrahamCampbell?

**Comment by julienfalque** [comment]
First case should be fixed by #6402.
